pkg/xds/cache/mesh/cache.go

```diff
@@ -15,12 +16,21 @@ import (
 // reconcile Dataplane's state. Calculating hash is a heavy operation
 // that requires fetching all the resources belonging to the Mesh.
 type Cache struct {
-	cache              *once.Cache
-	meshContextBuilder xds_context.MeshContextBuilder
+	// cache is used for caching a context and ignoring mesh changes for up to a
+	// short expiration time.
+	cache *once.Cache
+	// hashCache keeps a cached context, for a much longer time, that is only reused
+	// when the mesh hasn't changed.
+	hashCache *cache.Cache
 
-	latestMeshContext map[string]*xds_context.MeshContext
+	meshContextBuilder xds_context.MeshContextBuilder
 }
 
+// recalculationTime is the time after which the mesh context is recalculated no
+// matter what.
```

Comment:
Why do we need to recalculate no matter what?
Wouldn't it be better to "extend the longer cache lease" every time we compute that the mesh context did not change?

So the logic will be to always call `c.hashCache.SetDefault(mesh, meshCtx)` if we don't have an error after `c.meshContextBuilder.BuildIfChanged(ctx, mesh, context.Hash)`. This way we could
1) Change the expiration time of a longer cache to something shorter (like 1 minute?) causing deleted mesh contexts to be cleaned up faster
2) Avoid recomputing mesh context every hour if nothing changes

Reply:
Yeah that makes sense